app.js

```diff
@@ -0,0 +1,208 @@
+// this is the game board
+const gameBoard = () => {
+
+    const board = [0,1,2,3,4,5,6,7,8];
+    const winningCombos = [
+      [0, 1, 2],
+      [3, 4, 5],
+      [6, 7, 8],
+      [0, 3, 6],
+      [1, 4, 7],
+      [2, 5, 8],
+      [2, 4, 6],
+      [0, 4, 8]
+    ];
+
+    const setMark = (mark,index) => {
+      console.log('this here should work')
+      console.log(game.board[index]);
+      if(typeof game.board[index] == "number") {
+        game.board[index] = mark;
+        return true;
+       }
+       return false;
+      }
+
+    const getBoard = () => game.board;
+    const checkForWinner = (mark,currentPlayer)=> {
+      return winningCombos.some(combo => combo.every(e => currentPlayer.moves.includes(e)));
+    }
+
+    return { getBoard, setMark,checkForWinner, board };
+}
+const playerFactory = (name, mark) => {
+    const moves=[];
+    const play = (index,board) => {
+      a=game.setMark( this.mark,index);
+
+      if(a){
+        moves.push(index);
+        return a}
+        else{
+          return a
+        }
+
+    };
+    return{ name, mark, play, moves};
+  };
+const gameControlls = (game,grid,firstPlayer,secondPlayer) => {
+  let currentPlayer = firstPlayer;
+
+  const getCurrentPlayer = () => currentPlayer;
+  const setCurrentPlayer = (p) => currentPlayer = p;
```

Comment:
Again I'd like to repeat my previous query on this. Why do we need these `get` and `set` functions? Yeah I understand getters and setters. But I don't think for this cases we need them. For instance this `setCurrentPlayer` function simply assigns `currentPlayer` to be `p`. If you use a linter you will see that this code structure will raise an error.

Reply:
We removed them and just used the return, and setting current players in the if